### DISCUSSION TITLE
how does one define a "system-ui" font for qutebrowser?

### DISCUSSION THREAD

**Discussion by iambumblehead** [qutebrowser/qutebrowser#8365] [Q&A] [answered]
qutebrowser+gnome+postmarketos here. Some codeberg fonts look not-nice due to the way "system-ui" is rendered by qutebrowser. How can one change this?

<img src="https://github.com/user-attachments/assets/691a1606-8782-44fe-abd3-f1c06265f8f0" width="200px" />

system-ui font is used for "dnkl/yambar" and "code" and I want to change the appearance of those

```css
--fonts-proportional: -apple-system, "Segoe UI", system-ui, Roboto, "Helvetica Neue", Arial
```

qutebrowser here currently uses these font configurations,
```python
c.fonts.web.family.fixed = 'JetBrains Mono'
c.fonts.web.family.sans_serif = 'Cantarell'
c.fonts.default_family = ['JetBrains Mono']
```

thanks for any help

**Comment by toofar** [comment, marked as answer]
I'm pretty sure it just passes everything to fontconfig. If you do `fc-match system-ui` in a terminal it probably will show the same font? If you don't know what font is being used in the website you can select the element in the developer tools and look at the bottom of the "Computed" tab on the sidebar. Fontconfig will return a default font if there is no match, so if you don't have system-ui defined locally it probably returns the same thing as `fc-match asdf`.

**Reply by iambumblehead** [reply]
thanks, yes this is working with the following fontconfig/fonts.conf
```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
<fontconfig>
  <!-- Default font (no fc-match pattern) -->
  <match>
    <edit mode="prepend" name="family">
      <string>Noto Sans</string>
    </edit>
  </match>
  <!-- Default font for the ja_JP locale (no fc-match pattern) -->
  <match>
    <test compare="contains" name="lang">
      <string>ja</string>
    </test>
    <edit mode="prepend" name="family">
      <string>Noto Sans CJK JP</string>
    </edit>
  </match>
</fontconfig>
```
